source/opt/local_single_block_elim_pass.cpp

```diff
@@ -31,6 +31,10 @@ const uint32_t kStoreValIdInIdx = 1;
 bool LocalSingleBlockLoadStoreElimPass::HasOnlySupportedRefs(uint32_t ptrId) {
   if (supported_ref_ptrs_.find(ptrId) != supported_ref_ptrs_.end()) return true;
   if (get_def_use_mgr()->WhileEachUser(ptrId, [this](Instruction* user) {
+        auto dbg_op = user->GetOpenCL100DebugOpcode();
+        if (dbg_op == OpenCLDebugInfo100DebugDeclare) {
```

Comment:
@greg-lunarg I believe that jaebaek is right.  You should look for a DebugValue here as well.  Scalar replacement will be replacing DebugDeclare instructions with DebugValue instruction as shown above.  This would still be a supported ref.